File: l_m_s/features/lms/feature_lms/lib/models/ad_banner.dart
/// Domain model for ad/marketing banner (Sanity adBanner type).
class AdBanner {
  final String id;
  final String? headline;
  final String? imageUrl;
  final String? callToAction;
  final bool active;

  const AdBanner({
    required this.id,
    this.headline,
    this.imageUrl,
    this.callToAction,
    this.active = true,
  });

  factory AdBanner.fromMap(Map<String, dynamic> map) {
    String? imageUrl;
    final img = map['image'];
    if (img is Map && img['asset'] != null && img['asset']['url'] != null) imageUrl = img['asset']['url'] as String?;

    return AdBanner(
      id: map['_id'] as String? ?? '',
      headline: map['headline'] as String?,
      imageUrl: imageUrl,
      callToAction: map['callToAction'] as String?,
      active: map['active'] as bool? ?? true,
    );
  }
}

File: lms-studio/schemaTypes/adBanner.ts
import { defineType, defineField } from 'sanity'

export default defineType({
  name: 'adBanner',
  title: 'Ad / Marketing Banner',
  type: 'document',
  fields: [
    defineField({
      name: 'headline',
      title: 'Headline',
      type: 'string',
    }),
    defineField({
      name: 'description',
      title: 'Description',
      type: 'text',
    }),
    defineField({
      name: 'image',
      title: 'Image',
      type: 'image',
      options: { hotspot: true },
    }),
    defineField({
      name: 'callToAction',
      title: 'Call to Action',
      type: 'string',
      description: 'Button or link text (e.g. "Join Now")',
    }),
    defineField({
      name: 'active',
      title: 'Active',
      type: 'boolean',
      initialValue: true,
    }),
  ],
  preview: {
    select: { headline: 'headline', active: 'active' },
    prepare: ({ headline, active }) => ({
      title: headline || 'Banner',
      subtitle: active ? 'Active' : 'Inactive',
    }),
  },
})
